### PR TITLE
Fixed binary file generation problem in windows & Update `.gitignore`

### PR DESCRIPTION
**Pull request by C-BJ**
If don't specify the output file manually, the binary file will be generated without the `.exe` suffix in Windows

This PR mainly solves this problem

### CODE REVIEW

**Comment by C-BJ** [comment]
I don't know what's wrong with CI

**Comment by kyouko-taiga** [comment]
`main` is failing; working on it.
I'll re-run your PR once it's fixed.

**Comment by C-BJ** [comment]
I have cleaned all the commit

**Comment by C-BJ** [comment]
I have nothing to do with this CI failure(The CI failure I caused earlier has been resolved)

File: Sources/ValCommand/ValCommand.swift [hunk 1]
```diff
@@ -199,11 +199,18 @@ public struct ValCommand: ParsableCommand {
...
     let clang = try find("clang++")
-    let binaryURL = outputURL ?? URL(fileURLWithPath: productName)
+    var binaryPath = outputURL?.path ?? URL(fileURLWithPath: productName).path
+
+    #if os(Windows)
+      if !binaryPath.hasSuffix(".exe") {
+        binaryPath += ".exe"
+      }
+    #endif
+
```
Comment (kyouko-taiga) [inline Sources/ValCommand/ValCommand.swift:209]: This change would deserve to be factored out in a function so we can document the intent.

Comment (C-BJ) [reply Sources/ValCommand/ValCommand.swift:209]: Create a separate function?

File: .gitignore [hunk 2]
```diff
@@ -15,3 +15,5 @@
 /Sources/Parser/**
 !/Sources/Parser/Val.g4
 !/Sources/Parser/ValParser+Extensions.swift
+
+**.exe
```
Comment (kyouko-taiga) [inline .gitignore:19]:
```suggestion
# Windows
**.exe
```

Comment (kyouko-taiga) [reply .gitignore:19]: This section should come next to `macOS`, for consistency.

**Review by kyouko-taiga — CHANGES_REQUESTED** [review]
See inline comments.

File: Sources/ValCommand/ValCommand.swift (hunk 1 above)
Comment (kyouko-taiga) [reply Sources/ValCommand/ValCommand.swift:209]: Yes, exactly. Maybe:

```swift
/// Returns the path for an executable `name` at given `prefix`.
private func executablePath(_ prefix: URL, _ name: String) -> String {
  ...
}
```

**Comment by C-BJ** [comment]
Sorry, I did something wrong.

**Review by kyouko-taiga — APPROVED** [review]
Looks good, thanks.

**Pull request merged**